### DISCUSSION TITLE
Unhandled Exception: IsarError: Unique index violated.

### DISCUSSION THREAD

**Discussion by anandsubbu007** [isar/isar#255] [Help]
how to update data

**Comment by simc** [comment]
If you have a unique index, duplicate values are not allowed. Could you please share your collection class?

**Reply by anandsubbu007** [reply]
If i need to update that same item without changing that unique value. 
Is that also violate unique index?

**Reply by simc** [reply]
Not as long as both have the same id

**Reply by anandsubbu007** [reply]
Ok thanks

**Comment by kutear** [comment]
```
class Collection {
  Id id = Isar.autoIncrement;
  @Index(unique: true)
  late String url;
}
```
how can i use `putAll(...)` to insert unique index and discard duplicate index ?
